docs/index.md

```diff
@@ -0,0 +1,38 @@
+## Introduction
+
+You can learn from this guide how to use OAuth via a Google server to:
+
+- [connect to Settle](Connecting_to_Settle.md)
+- [accept connections from Settle](Accepting_connections_from_Settle.md)
+
+We are using Google Cloud Platform (GCP) service accounts and
+authorization servers to secure endpoints in our APIs.
+
+We propose that you use the same principles to secure endpoints in our APIs.
+
+After reading the docs for how to connect, consider taking a look at
+[the Python demo for the entire flow](python_demo.md).
+
+## The OAuth flow
+
+This is a generic diagram that applies in both directions, i.e. for both
+connecting to settle and for accepting connections from Settle.
+Please refer back to this for clarity on terminology and discrete steps
+if any confusion arises while reading the other docs.
+
+```mermaid
+sequenceDiagram
+    participant C as Client Service
+    participant G as GCP Auth Server
+    participant S as API Server
+    C ->>+ G: SvcAcctKey, Audience
+    Note right of C: Audience = Server endpoint URL
```

Comment:
Not sure there's necessarily a 1:1 correspondence here in our APIs, but maybe there should be?

Reply:
This is a convention we want to enforce because it provides the best security at almost no cost to us or the partners. Our API server code would enforce this.

Reply:
So IIUC this means the client is to maintain one token per API endpoint, instead of one token per API?

Reply:
They obtain the token automatically via the client library from the auth server, they don't maintain anything, there's no hassle.

Reply:
Ok. Probably we should discuss this to improve my understanding of the consequences.